src/Tests/TestComponentCSharp/Class.cpp

```diff
@@ -1553,6 +1558,67 @@ namespace winrt::TestComponentCSharp::implementation
         return winrt::make<bindable_observable_vector>(vector);
     }
 
+    bool Class::ValidateBindableProperty(
+        WF::IInspectable const& bindableObject,
+        hstring property,
+        Windows::UI::Xaml::Interop::TypeName const& indexerType,
+        bool validateOnlyExists,
+        bool canRead,
+        bool canWrite,
+        bool isIndexer,
+        Windows::UI::Xaml::Interop::TypeName const& type,
+        WF::IInspectable const& indexerValue,
+        WF::IInspectable const& setValue,
+        WF::IInspectable& retrievedValue)
+    {
+        auto customPropertyProvider = bindableObject.as<ICustomPropertyProvider>();
+        auto customProperty = !isIndexer ? customPropertyProvider.GetCustomProperty(property) : customPropertyProvider.GetIndexedProperty(property, indexerType);
+        if (customProperty == nullptr)
+		{
+            return false;
+		}
+
+        if (validateOnlyExists)
+		{
+			return true;
+		}
+
+        if (customProperty.Name() != property ||
```

Comment:
what's the purpose of passing these bools in?  given they have to match customProperty, why not just use the latter? (they seem redundant)

Reply:
This is validating expected vs actual.  I will have a follow up PR to address this along with your enum comment for easier readability.